### PR TITLE
[RTM] Fixed DEPRECATED.md description and related code changes

### PR DESCRIPTION
**Pull request by Toflar**

### CODE REVIEW

File: src/EventListener/InitializeSystemListener.php [hunk 1]
```diff
@@ -138,10 +138,12 @@ public function onConsoleCommand()
     protected function setConstants($mode, $route)
     {
         // The constants are deprecated and will be removed in version 5.0.
+        $time = microtime(true);
```
Comment (leofeyer) [inline src/EventListener/InitializeSystemListener.php]: Please revert this change. It has nothing to do with the `DEPRECATED.md` file.

File: DEPRECATED.md [hunk 2]
```diff
@@ -15,7 +15,36 @@ You can use the static helper methods such as `System::loadLanguageFile()` or
 The constants `TL_ROOT`, `TL_MODE`, `TL_START` and `TL_SCRIPT` have been
 deprecated and will be removed in Contao 5.0.
 
-// FIXME: show the alternatives
+Instead of `TL_ROOT` use:
+
+```php
+global $kernel;
+
+$rootDir = $kernel->getContainer()->getParameter('kernel.root_dir');
```
Comment (aschempp) [inline DEPRECATED.md]: This will not give the same value, you need to add a `dirname()`

File: DEPRECATED.md [hunk 3]
```diff
@@ -15,7 +15,36 @@ You can use the static helper methods such as `System::loadLanguageFile()` or
...
+Instead of `TL_MODE` to check for `BE` and `FE` use:
+
+```php
+global $kernel;
+
+$isFE = $kernel->getContainer()->isScopeActive('frontend');
+$isBE = $kernel->getContainer()->isScopeActive('backend');
+```
+
+Note: You can also use the `ScopeAwareListener` in your own event listeners.
```
Comment (aschempp) [inline DEPRECATED.md]: What do you mean by that?

File: DEPRECATED.md [hunk 4]
```diff
@@ -15,7 +15,36 @@ You can use the static helper methods such as `System::loadLanguageFile()` or
...
+Instead of `TL_START` use:
+```php
+global $kernel;
+
+$startTime = $kernel->getStartTime();
+```
+
+Note: This will only work if Symfony is in debug mode (`kernel.debug`).
+
+Instead of `TL_SCRIPT` use the Symfony routing component to generate proper URLs.
```
Comment (leofeyer) [inline DEPRECATED.md]: Aren't people mainly using `TL_SCRIPT` to check for the current entry point?

```php
if (TL_SCRIPT == 'contao/main.php') {
}
```

We should probably describe both cases, shouldn't we?

File: DEPRECATED.md (hunk 3 above)
Comment (Toflar) [inline DEPRECATED.md]: I mean what is written. You can use the `ScopeAwareListener` in your own event listeners if you like.

File: DEPRECATED.md (hunk 4 above)
Comment (Toflar) [inline DEPRECATED.md]: No idea, I never ever used `TL_SCRIPT`.

File: src/EventListener/InitializeSystemListener.php (hunk 1 above)
Comment (Toflar) [inline src/EventListener/InitializeSystemListener.php]: Well, we should't use `TL_START` either if we're deprecating it ;-)

Comment (leofeyer) [inline src/EventListener/InitializeSystemListener.php]: So let's just add a second example (assuming the following code is correct):

```php
if ('contao_backend_main' === $this->container->get('request')->get('_route')) {
}
```

File: DEPRECATED.md (hunk 4 above)
Comment (leofeyer) [inline DEPRECATED.md]: So let's just add a second example (assuming that the following code is correct):

```php
if ('contao_backend_main' === $this->container->get('request')->get('_route')) {
}
```

File: src/EventListener/InitializeSystemListener.php (hunk 1 above)
Comment (leofeyer) [inline src/EventListener/InitializeSystemListener.php]: If `TL_REFERER_ID` is **not** deprecated, I agree.

Comment (Toflar) [inline src/EventListener/InitializeSystemListener.php]: It's not (yet).

File: DEPRECATED.md (hunk 4 above)
Comment (Toflar) [inline DEPRECATED.md]: Better?

File: DEPRECATED.md (hunk 2 above)
Comment (Toflar) [inline DEPRECATED.md]: Fixed 136a52740cd1a0977bf46d4ce7c3c640d11bd56f.

File: src/EventListener/InitializeSystemListener.php (hunk 1 above)
Comment (discordier) [inline src/EventListener/InitializeSystemListener.php]: But should be.
IMO we should deprecate every constant to encourage writing properly testable code.
Constants not be used anymore not by legacy extensions and not even in the core-bundle but this will take a migration phase during the 4.x series.
To allow this, we should deprecate them all as of now.

Comment (Toflar) [inline src/EventListener/InitializeSystemListener.php]: Of course it should be but we cannot deprecate anything as long as there's no alternative to be used :)

Comment (leofeyer) [inline src/EventListener/InitializeSystemListener.php]: And there is no alternative to the referer ID yet.

Comment (Toflar) [inline src/EventListener/InitializeSystemListener.php]: I'll try to find one.

Comment (leofeyer) [inline src/EventListener/InitializeSystemListener.php]: Ok. In the meantime, please revert this change, since it is not related to the `DEPRECATED.md` file.

**Comment by Toflar** [comment]
Reverted changes in c8fd4cffaaf0f02a67c65d74a93b074d36efb094.
